bluetooth: fix le addr invlaid

chengkai15 opened the pull request:
*Note: Please adhere to [Contributing Guidelines](https://github.com/open-vela/docs/blob/dev/CONTRIBUTING.md).*

## Summary

1. [bluetooth: fix le addr invlaid](https://github.com/open-vela/frameworks_bluetooth/commit/1acb2fbe3e7947bedf45d3881307471b7e279608)
2. [bluetooth: fix SCAN_TYPE build break](https://github.com/open-vela/frameworks_bluetooth/commit/1a3bdd9f1f2d6a76ffff43872b8f35cea5db5655)

## Impact

get le addr

## Testing

get leaddr pass

chengkai15 committed:
bluetooth: fix SCAN_TYPE build break

bug: v/55753

Signed-off-by: chengkai <chengkai@xiaomi.com>

chengkai15 committed:
bluetooth: fix le addr invlaid

bug: v/55753

Signed-off-by: chengkai <chengkai@xiaomi.com>

gzh-terry commented on service/stacks/zephyr/sal_adapter_le_interface.c:
```diff
@@ -613,10 +613,19 @@ bt_status_t bt_sal_le_set_address(bt_controller_id_t id, bt_address_t* addr)
+
+    SAL_CHECK_PARAM(addr);
+
+    bt_id_get(&got, &count);
```
overflow risk here?

```C
void bt_id_get(bt_addr_le_t *addrs, size_t *count)
{
	if (addrs) {
		size_t to_copy = MIN(*count, bt_dev.id_count);

		memcpy(addrs, bt_dev.id_addr, to_copy * sizeof(bt_addr_le_t));
		*count = to_copy;
	} else {
		*count = bt_dev.id_count;
	}
}
```

chengkai15 committed:
bluetooth: fix bredr noinputoutput pair fail

bug: v/46022

rootcause: noinputoutput do not meet unauthenticated request, then
set to level l2.

Signed-off-by: chengkai <chengkai@xiaomi.com>

huangyulong3 reviewed (approved):
LGTM.

huangyulong3 commented on service/src/adapter_service.c:
```diff
@@ -1149,10 +1149,20 @@ void adapter_on_le_enabled(bool enablebt)
 #ifdef CONFIG_BLUETOOTH_BLE_SUPPORT
     adapter_service_t* adapter = &g_adapter_service;
     int ret;
+    char addrstr[BT_ADDR_STR_LENGTH];
```
Here the length needs to be changed to` BT_ADDR_STR_LENGTH + 1` and initialized `addrstr` array to 0 , otherwise the `BT_LOGD(“%s, le_addr:%s”, __func__, addrstr)` function later on will print gibberish. If it is in qemu/sim environment there will be read out of bounds crash.

chengkai15 replied:
BT_ADDR_STR_LENGTH  was  already 18 bytes ，which is one more size for address

This is redundant

hyson710 commented on framework/include/bt_le_scan.h:
```diff
@@ -22,6 +22,10 @@ extern "C" {
 #include <stdint.h>
 
+#ifdef CONFIG_BLUETOOTH_STACK_LE_ZBLUE
+#include <zephyr/bluetooth/bluetooth.h>
```
In the future, we need to fix the conflicts in SAL layer

chengkai15 replied:
OK
